networking/podenv.go

```diff
@@ -154,6 +173,25 @@ func (e *podEnv) podNSPathSave() error {
 	return nil
 }
 
+func (e *podEnv) podNSDestroy() error {
+	if e.podNS == nil {
+		return nil
+	}
+	_ = e.podNS.Close() // Don't care if err here
+
+	if err := syscall.Unmount(e.podNS.Path(), unix.MNT_DETACH); err != nil {
+		// if already unmounted, umount(2) returns EINVAL
+		if !os.IsNotExist(err) && err != syscall.EINVAL {
```

Comment:
A follow up.
1. I tested, we do not need to cast to errno
2. This is only reachable if someone steps on our toes and deletes the NS during execution.